### PR TITLE
Fix child transmission target can't inherit Parent transmission target for common property

### PR DESCRIPTION
**Pull request by AnastasiaKubova**
Things to consider before you submit the PR:

* [ ] ~Has `CHANGELOG.md` been updated?~
* [ ] ~Are tests passing locally?~
* [x] Are the files formatted correctly?
* [ ] ~Did you add unit tests if this modifies the Windows code?~
* [x] Did you test your change with either the sample apps that are included in the repository or with a blank app that uses your change?

## Description

Repro Steps

1. Open Unity Demo app
2. Select any Startup mode and restart demo app
3. Click Transmission target enter Event Name
4. Click Add Property button to add property "A" for Parent target
5. Then click Add Property button to add property "B" for Child target
6. Observe the result on Aria portal if the property "A" can be inherited to Child target

## Related PRs or issues

[AB#69752](https://msmobilecenter.visualstudio.com/454a20a9-dfe6-4c1f-8ae8-417f76adb472/_workitems/edit/69752)

### CODE REVIEW

**Review by annakocheshkova — DISMISSED** [review]
Well, Looks good to me 😉

**Review by olmisa — DISMISSED** [review]

File: Assets/Puppet/PuppetTransmission.cs [hunk 1]
```diff
@@ -395,82 +408,88 @@ public void TrackEventParentTypedPropertiesTransmission()
...
-                foreach (var property in properties)
-                {
-                    propertyConfigurator.SetEventProperty(property.Key, property.Value);
-                }
-                propertyConfigurator.SetEventProperty("extraEventProperty", "should be removed!");
-                propertyConfigurator.RemoveEventProperty("extraEventProperty");
+                var propertyConfigurator = ApplyPropertyToString(childTransmissionTarget, EventChildPropertiesList);
                 if (_isCritical)
                 {
+                    System.Diagnostics.Debug.WriteLine("TrackEventStringPropertiesChildTransmission _isCritical");
```
Comment (MatkovIvan) [inline Assets/Puppet/PuppetTransmission.cs]: Remove temporary log, or replace to Unity's `Debug.Log`

File: Assets/Puppet/Utility/PropertiesHelper.cs [hunk 2]
```diff
@@ -52,6 +52,10 @@ public static void AddPropertiesToPropertyConfigurator(RectTransform propertiesC
 
     public static void RemovePropertiesFromConfigurator(RectTransform propertiesContainer, PropertyConfigurator configurator)
     {
+        if (configurator == null)
+        {
+            return;
```
Comment (MatkovIvan) [inline Assets/Puppet/Utility/PropertiesHelper.cs:58]: Why it's the case? Should we log something here?

**Review by MatkovIvan — CHANGES_REQUESTED** [review]

**Review by annakocheshkova — DISMISSED** [review]

**Review by ilyashumihin — DISMISSED** [review]

**Review by Jamminroot — DISMISSED** [review]

**Review by Cloudwu1 — DISMISSED** [review]

**Review by MatkovIvan — DISMISSED** [review]

**Review by guperrot — DISMISSED** [review]

File: Assets/Puppet/Utility/PropertiesHelper.cs [hunk 3]
```diff
@@ -52,6 +52,11 @@ public static void AddPropertiesToPropertyConfigurator(RectTransform propertiesC
 
     public static void RemovePropertiesFromConfigurator(RectTransform propertiesContainer, PropertyConfigurator configurator)
     {
+        if (configurator == null)
+        {
+            Debug.Log("Property configurator is null. Can not remove properties from it.");
```
Comment (jaeklim) [inline Assets/Puppet/Utility/PropertiesHelper.cs]:
```suggestion
            Debug.Log("Property configurator is null. Can't remove properties from it.");
```

File: Assets/Puppet/PuppetTransmission.cs [hunk 4]
```diff
@@ -312,6 +312,28 @@ public void ClearChildUserId()
         OnChildUserIdChanged(null);
     }
 
+    private PropertyConfigurator ApplyPropertyToString(TransmissionTarget transmissionTarget, RectTransform eventProperties)
```
Comment (jaeklim) [inline Assets/Puppet/PuppetTransmission.cs]: Please reconsider the method name.

File: Assets/Puppet/PuppetTransmission.cs [hunk 5]
```diff
@@ -312,6 +312,28 @@ public void ClearChildUserId()
...
+        foreach (var property in properties)
+        {
+            propertyConfigurator.SetEventProperty(property.Key, property.Value);
+        }
+        propertyConfigurator.SetEventProperty("extraEventProperty", "should be removed!");
+        propertyConfigurator.RemoveEventProperty("extraEventProperty");
+        return propertyConfigurator;
+    }
+
+    private PropertyConfigurator ApplyPropertyToType(TransmissionTarget transmissionTarget, RectTransform eventProperties)
```
Comment (jaeklim) [inline Assets/Puppet/PuppetTransmission.cs]: Please reconsider the method name.

**Review by thyeggman — DISMISSED** [review]

**Review by annakocheshkova — APPROVED** [review]

**Review by guperrot — APPROVED** [review]

**Pull request merged**